File: app/src/main/java/com/juniperphoton/myersplash/utils/ColorUtil.java
package com.juniperphoton.myersplash.utils;

import android.graphics.Color;

public class ColorUtil {
    public static boolean isColorLight(int color) {
        double luma = getLumaFromColor(color);
        return luma >= 120;
    }

    private static double getLumaFromColor(int color) {
        int r = Color.red(color);
        int g = Color.green(color);
        int b = Color.blue(color);
        return 0.299 * r + 0.587 * g + 0.114 * b;
    }

    public static int getDarkerColor(int color, float alpha) {
        return Color.argb((int) (255 * alpha), Color.red(color), Color.green(color), Color.blue(color));
    }
}

File: app/src/main/java/com/juniperphoton/myersplash/utils/ColorUtil.kt
package com.juniperphoton.myersplash.utils

import android.graphics.Color

class ColorUtil {
    companion object {
        fun isColorLight(color: Int): Boolean {
            return getLumaFromColor(color) > 120
        }

        private fun getLumaFromColor(color: Int): Double {
            return 0.299 * Color.red(color) + 0.587 * Color.green(color) + 0.114 * Color.blue(color)
        }

        fun getDarkerColor(color: Int, alpha: Float): Int {
            return Color.rgb((Color.red(color) * alpha).toInt(), (Color.green(color) * alpha).toInt(),
                    (Color.blue(color) * alpha).toInt())
        }
    }
}